### PR TITLE
query optimization

### PR DESCRIPTION
**Pull request by happythenewsad**
## WHAT
Adds materialized view for FeedbackHistory, allowing us to optimize the Rules Report query

## WHY
Rules Report times out

### Screenshots
(If applicable. Also, please censor any sensitive data)

### Notion Card Links
https://www.notion.so/quill/Product-Board-30f97e4eb01246dbb8264253fb385073?p=0c2087d11d364fbc817e65fdccb49b87

PR Checklist | Your Answer
------------ | -------------
Have you added and/or updated tests? |  (The answer should mostly be 'YES'. If you answer 'NO', please justify.)
Have you deployed to Staging? | yes
Self-Review: Have you done an initial self-review of the code below on Github? |
Design Review: If applicable, have you compared the coded design to the mockups? | (N/A or Yes)

### CODE REVIEW

File: services/QuillLMS/db/structure.sql [hunk 1]
```diff
@@ -452,6 +452,47 @@ CREATE SEQUENCE public.activity_classifications_id_seq
 ALTER SEQUENCE public.activity_classifications_id_seq OWNED BY public.activity_classifications.id;
 
 
+--
+-- Name: activity_healths; Type: TABLE; Schema: public; Owner: -
+--
+
+CREATE TABLE public.activity_healths (
```
Comment (happythenewsad) [inline services/QuillLMS/db/structure.sql:459]: These were not related to this PR, but were not versioned before. I checked with Cissy that it's ok to include these structure.sql changes in this PR.

**Comment by happythenewsad** [comment]
The build fails due to an intermittently-failing spec. Disregard

**Review by emilia-friedberg — APPROVED** [review]

**Review by anathomical — APPROVED** [review]
This looks good.  Do we need to schedule some sort of job to refresh the view?  Looks like the view will go stale pretty quickly with the current configuration.  Unless I missed something that's triggering refreshes?

**Comment by happythenewsad** [comment]
> This looks good. Do we need to schedule some sort of job to refresh the view? Looks like the view will go stale pretty quickly with the current configuration. Unless I missed something that's triggering refreshes?

Good point - I'm planning to deploy this, observe the performance, and then add a chron for the refresh, with a reasonable interval. I should have mentioned this in the WHAT section. CC @emilia-friedberg as well in case she had the same question

**Pull request merged**